cvejob/version_utils.py

```diff
@@ -0,0 +1,336 @@
+"""Module for processing and interpreting NVD version ranges.
+
+The easiest way how to use this module is call `get_ranges_from_cve()`
+function.
+"""
+
+import nvdlib.utils
+from cpe import CPE
+from enum import Enum
+import re
+
+from f8a_version_comparator.comparable_version import ComparableVersion
+from cvejob.version import BenevolentVersion
+from f8a_utils.versions import (
+    get_versions_for_maven_package, get_versions_for_pypi_package, get_versions_for_npm_package
+)
+
+
+def get_ranges_from_cve(cve, package, ecosystem):
+    """Get version ranges for given CVE.
+
+    :param cve: nvdlib.model.CVE, CVE object
+    :param package: str, package name
+    :param ecosystem: str, ecosystem name
+
+    :return: list of affected version ranges, list of unaffected version ranges
+    """
+    nodes = get_configuration_nodes(cve)
+    cve_ranges = []
+    for node in nodes:
+        for version_range in node:
+            version_ranges = get_version_ranges(version_range)
+            if version_ranges:
+                cve_ranges.append(VersionRange.from_range_list(version_ranges))
+    upstream_versions = get_upstream_versions(package, ecosystem)
+    classified_versions = classify_versions(upstream_versions, cve_ranges)
+    affected = get_affected_ranges(classified_versions)
+    safe = get_safe_ranges(classified_versions)
+    return affected, safe
+
+
+class ClassifiedVersion(object):
+    """Version classified as affected or unaffected.
+
+    TODO: if Python 3.7: turn into a data class
```

Comment:
Dataclasses have been backported to Python 3.6: https://pypi.org/project/dataclasses/
`pip install dataclasses`

Reply:
Ah, nice! But let's wait for Python 3.7. I don't want to add another external dependency to the project.